### ISSUE TITLE
Deploying non-framework TypeScript repo results in 0 Serverless functions being detected, and ignored static files being served

### ISSUE THREAD

**Issue by chriseagle** [zeit/now#9822]
**tl:dr** Deploying with a build step and a configured outputDirectory results in 0 serverless functions being detected. Deploying with a build committed and built to root of repo works correctly.

Building a non-framework based TypeScript scaffold to deploy on Vercel, the project it using tsup and is currently very bare. When building as part of the deploy and configuring the outputDirectory in GUI and via `vercel.json` results in a build output of **0 Serverless f**unctions discovered.

The build outputs to `./dist` and outputDirectory is configured to "dist". 

#### Directory Structure
```
src
-api
--handler.ts
dist
-api
--handler.js
```

#### Vercel.json
```
{
    "outputDirectory": "dist",
}
```

### Other things I've tried that don't work

1. Switching to building before deploy and removing the build step, building to a sub directory still fails. 
2. Defining `framework` as "Other" or "" prevents deploy.

### Workarounds

1. Switching to building before deploy and building to root with a default `outputDirectory` works, however static files are detected including those which are part of `.vercelignore`.

#### Directory Structure for the above workaround 
```
src
-api
--handler.ts
-api
--handler.js
```
`./src` is added to .vercelignore.

### Expected behaviour

I would expect `outputDirectory` to function so the provided dir is treated as the root of the project to be served and for all files listed in .vercelignore to not be available. 

### Below screenshot is example of an ignored file being statically served.

![Screenshot 2023-04-17 at 15 58 52](https://user-images.githubusercontent.com/4096392/232525318-b7e6b1ef-b5e7-47fa-8ac9-8307b5a475af.png)
